wandb/sdk/artifacts/storage_handlers/s3_handler.py

```diff
@@ -93,13 +93,18 @@ def load_path(
 
         extra_args = {}
         if version:
-            obj = self._s3.ObjectVersion(bucket, key, version).Object()
+            obj_version = self._s3.ObjectVersion(bucket, key, version)
             extra_args["VersionId"] = version
+            obj = obj_version.Object()
         else:
             obj = self._s3.Object(bucket, key)
 
         try:
-            etag = self._etag_from_obj(obj)
+            etag = (
```

Comment:
line 124 looks wrong i think. You probably want to use the `etag_from_obj_version` call instead of the old one

Reply:
Both methods seem to work on the objectVersion resource returned by `object_versions()`. I think the APIs are pretty inconsistent with the metadata they return with the ObjectVersion resources. So that can stay as is.

Reply:
I guess it is really confusing to read. If both method work, there is no reason to create the new `_etag_from_obj_version` method. We should just use the old method, otherwise, we should use the correct method for the correct type. Technically, line 124 shouldnt pass type check right? What is the return type of `self._s3.Bucket(bucket).object_versions.filter(Prefix=key)`? It looks like an `ObjectVersion` to me

Reply:
This seems to be due to an issue with the underlying S3 API. See here: https://github.com/boto/boto3/issues/832

Ideally, for all `ObjectVersion` instances, we should be able to retrieve the etag in the same way as we do for an `Object`. But based on what I've seen and what's described in the issue, unless the ObjectVersion is a part of a `collection` resource, it doesn't have the etag metadata already and has to 'load' it. 

I think changing the new method name to `load_etag_from_obj()` will make the code clearer.

Reply:
Nevermind a new method. I updated `_etag_from_obj()` to handle the extra data load if needed.